### PR TITLE
Expose smart punctuation options on CupertinoSearchTextField

### LINKED ISSUE #102058: [Proposal] Allow disabling smart-quotes in CupertinoSearchTextField
On `CupertinoTextField` you can set `smartQuotesType` so that the `"` doesn't get turned into `“`. It would be nice to have the same option exposed on `CupertinoSearchTextField` to avoid having to normalize before using the search field contents.

### PR DESCRIPTION
**Pull request by moffatman**
Let users of `CupertinoSearchTextField` choose to disable automatic directional quotes, en-dashes, and em-dashes.

Fixes #102058

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/97980

**Review by guidezpl — CHANGES_REQUESTED** [review]
Not at all familiar with this mechanism, but this needs a test to validate behavior.

**Comment by moffatman** [comment]
@guidezpl There is a test here, it's just a change to pass-through an option to a child Widget.

**Review by guidezpl — APPROVED** [review]
I shouldn't review on my phone :). LGTM

**Comment by guidezpl** [comment]
Running Google testing again

**Comment by moffatman** [comment]
@guidezpl Can you press the merge button?

**Comment by moffatman** [comment]
@guidezpl Guess it needs 1 more approval

**Comment by guidezpl** [comment]
Can you post a link in the Discord for a review?

File: packages/flutter/lib/src/cupertino/search_field.dart
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/search_field.dart:276]: The property is nullable so maybe explain what null means too.

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/cupertino/search_field.dart
Comment (moffatman) [reply packages/flutter/lib/src/cupertino/search_field.dart]: Actually looked into it, the default platform behavior is never used. It will either be forced to yes or no. But I updated the comment to explain that.

**Comment by moffatman** [comment]
@guidezpl  Can you now press the merge button?

**Pull request merged**
